test/e2e_v2.t.sol

```diff
@@ -36,6 +37,7 @@ contract TestEndToEndDeploymentV2 is Test {
     function setUp() public {
         deployFactoryV2 = new DeployFactoryV2();
         deployTokenV2 = new DeployTokenV2();
+        tokenFactory = new TokenFactoryV2{salt: keccak256("TOKEN_FACTORY_SALT")}(owner);
```

Comment:
last change:
instead of deploying the contract via the contract, deploy it via the script.
like this: `TokenFactoryV2 tokenFactory = deployFactoryV2._run(owner);`

Reply:
so delete this line